### ISSUE TITLE
3 classes are unsatisfiable in SWEET 2.3

### ISSUE THREAD

**Issue by cmungall** [ESIPFed/sweet#9]
Hi, I work with @pbuttigieg on ENVO

I note that the disjointness axiom between 'scalar' and 'vector' in reprMath is causing three classes to become unsatisfiable:

 - Horizontal Coordinate
 - Medirian
 - PrimeMeridian

I can't seem to get an explanation to show in Protege for some reason, so a textual exemplar may have to suffice:

 - 'orthogonal to' has domain Direction, Direction isa 3D Vector isa Vector
 - HorizontalCoordinate isa Coordinate isa Scalar
 - Scalar DisjointFrom Vector

**Comment by lewismc** [comment]
Hi @cmungall thanks for registering the inconsistency/bug. I have not looked at this yet, but will do so in due course. As part of the transition of SWEET to ESIPFed, we are also looking at who is capable of being a subject matter expert(s) for various aspects of the suite.

**Comment by lewismc** [comment]
@cmungall  see #3

**Comment by cmungall** [comment]
With what time I have available I will contribute what I can to the biological phenomena branch, and to general cross-domain issues of involving OWL/reasoning/standards/data integration/etc.

**Comment by lewismc** [comment]
ack @cmungall we have our monthly meetings as follows
```
Monthly Telecon

    4th Tuesday of each month at 4pm Eastern
    GoToMeeting: https://www.gotomeeting.com/join/976796333
    Phone Access: United States: +1 (872) 240-3212
    Access Code: 976-796-333 
```
If you are interested in joining then please do.

**Comment by lewismc** [comment]
@cmungall I wonder, are you able to revisit this? I still do not really understand much of it.

**Comment by cmungall** [comment]
On 30 Aug 2017, at 14:00, Lewis John McGibbney wrote:

> @cmungall I wonder, are you able to revisit this? I still do not 
> really understand much of it.


It seems to be fixed in the latest version of sweet in github

**Comment by lewismc** [comment]
ack, thank you for reporting

**Comment by cmungall** [comment]
OK, I went back and had a look again. I cannot reproduce directly the inconsistency.

However, this axiom is extremely dubious:

`[1] HorizonalCoordinate SubClassOf orthogonalTo only VerticalCoordinate`

orthogonalTo is constrained to be between Vectors, Coordinates are Scalars.

Note that this axiom will not cause an inconsistency because it's quantified with 'only'. It's like saying:

`Dog SubClassOf has-paid-job only some BusDriver`
`Cat SubClassOf has-paid-job only some AirTrafficController`

This is trivially true: the total set of all dog occupations is the empty set, which is a subset of the set of all bus drivers.

However, no ontology would intentionally have an axiom like this.

I think it's the same for [1], it's axiomatically silent so long as there are no horizontal coordinates with the orthogonalTo property. The moment there are, the ontology becomes inconsistent.

To demonstrate I added 3 axioms

 1. `TestHorizontalCoordinate a HorizontalCoordinate`
 2. `TestVerticalCoordinate a VerticalCoordinate`
 2. `TestHorizontalCoordinate orthogonalTo TestVerticalCoordinate`

This cause an inconsistency, as demonstrated in this explanation:

![image](https://user-images.githubusercontent.com/50745/29932857-5e188148-8e2a-11e7-9a5f-d65221696222.png)

TL;DR - I think the intent is to place the universal axiom on Horizontal*Direction* not Horizontal*Coordinate*

**Comment by lewismc** [comment]
This is much clearer @cmungall thank you for clarifying your understanding.
If wonder however if we could validate the following statement

> TL;DR - I think the intent is to place the universal axiom on HorizontalDirection not HorizontalCoordinate

Can we test this using HorizontalDirection via the same logic you've applied to HorizontalCoordinate ?

**Comment by graybeal** [comment]
I think his test will work fine if the Directions are used for both (Horizontal.. and Vertical..). But I'm wondering about the meaning, and specifically whether 'only' is appropriate. Can it go away?

**Comment by cmungall** [comment]
> I think his test will work fine if the Directions are used for both (Horizontal.. and Vertical..).

correct, but not a bad idea to check anyway, just in case I'm full of it

> But I'm wondering about the meaning, and specifically whether 'only' is appropriate. Can it go away?

This is where it helps to document high level design decisions. Someone at some point decided it was a good idea to add these to sweet, but it's not clear (to me) who or why.

In this particular case, I think there may be some some justification for using universal restrictions (`only`) as a constraint mechanism. In general constraints are good (provided they are valid) as they serve as both regression tests in the ontology and as ways of checking the validity of data. If someone provides data (abox) with two parallel vectors and declares them to be orthogonal we'd want to detect this.

HOWEVER, even if you follow my original instructions and make the restriction between Horizontal and Vertical *Directions*:

```
HorizonalDirection SubClassOf orthogonalTo only VerticalDirection
```

I don't think this is quite right. It's hard to be sure as 'Horizontal' is not defined. It may seem obvious but it helps to be explicit especially when dealing with the intersection of description logics and mathematics.

I believe it's possible to have two horizontal directions that are mutually perpendicular. E.g. North and South (these are instances in SWEET). These are not vertical directions (such as up). Thus it's incorrect to state that every horizontal direction is orthogonal to only vertical directions.

I do think it's OK to say VerticalDirection SubClassOf orthogonalTo only HorizonalDirection.

This is a simplification as I think we're assuming approximation to planes here, but I think the analysis holds even for spheres, although I confess thinking about the intersection of OWL and non-euclidean geometry makes my head hurt.

Added onto this is the fact that Horizontal is itself really a relation..

**Comment by cmungall** [comment]
I'm not really sure I understand the use case for representing mathematical constructs and relations (vectors, orthogonality) in a DL ontology. We already have a good language for representing mathematical constructs - mathematics.

But assuming there is a use case, it might be fun to try and come up with axioms that are valid and as complete as possible.

We have a spatial ontology for biological entities:

http://obofoundry.org/ontology/bspo.html
https://jbiomedsem.biomedcentral.com/articles/10.1186/2041-1480-5-34

there may be similarities. E.g. everything is relative to an organism (or in sweet's case, a roughly spherical astronomical body).

I'm embarrassed as I see issues with our own definition of orthogonal-to:
http://purl.obolibrary.org/obo/BSPO_0015005

The other relations (e.g. superior-to, anterior-to etc) are better. We use these a lot in anatomical ontologies to state relative positioning of structures.

For sweet I think the best start is to start with more precise definitions of classes like horizontal, plus some use cases.
